scripts/build_bolt.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+# This builds an optimized version of the binary with bolt
+#
+# The profile data is captured by binary instrumentation via bolt.
+# You can find a version of file where the profile is captured using perf on Github in PR 2724
+# In our experience, instrumentation resulted in similar performance and was more portable.
+# This file needs to be executed inside a cmake build folder. We recommend having a separate folder
+# as this file changes the cmake configuration.
+#
+# For a usage string, run this binary with -h
+
+build_folder=$(pwd)
+os="$(uname -s)"
+case "${os}" in
+  Linux*)   num_cores="$(lscpu -p | grep -Ev '^#' | grep '^[0-9]*,[0-9]*,0,0' | sort -u -t, -k 2,4 | wc -l)";;
+  Darwin*)  num_cores="$(sysctl -n hw.physicalcpu)";;
+esac
+time_per_benchmark=1800
```

Comment:
Comment to tell that we recommend profiling for a decent amount of time but script can run much faster for CI when passing -t